livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java

```diff
@@ -0,0 +1,164 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.livy;
+
+import org.apache.zeppelin.interpreter.*;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+
+/**
+ * Livy PySpark interpreter for Zeppelin.
+ */
+public class LivySparkSQLInterpreter extends Interpreter {
+
+  Logger LOGGER = LoggerFactory.getLogger(LivySparkSQLInterpreter.class);
+  static String DEFAULT_MAX_RESULT = "1000";
+
+  static {
+    Interpreter.register(
+        "sql",
+        "livy",
+        LivySparkSQLInterpreter.class.getName(),
+        new InterpreterPropertyBuilder()
+            .add("livy.spark.maxResult",
+                DEFAULT_MAX_RESULT,
+                "Max number of SparkSQL result to display.")
+            .build()
+    );
+  }
+
+  protected Map<String, Integer> userSessionMap;
+  private LivyHelper livyHelper;
+
+  public LivySparkSQLInterpreter(Properties property) {
+    super(property);
+    livyHelper = new LivyHelper(property);
+    userSessionMap = LivySparkInterpreter.getUserSessionMap();
+  }
+
+  @Override
+  public void open() {
+  }
+
+  @Override
+  public void close() {
+    livyHelper.closeSession(userSessionMap);
+  }
+
+  @Override
+  public InterpreterResult interpret(String line, InterpreterContext interpreterContext) {
+    try {
+      if (userSessionMap.get(interpreterContext.getAuthenticationInfo().getUser()) == null) {
+        try {
+          userSessionMap.put(
+              interpreterContext.getAuthenticationInfo().getUser(),
+              livyHelper.createSession(
+                  interpreterContext,
+                  "spark")
+          );
+          livyHelper.initializeSpark(interpreterContext, userSessionMap);
+        } catch (Exception e) {
+          return new InterpreterResult(InterpreterResult.Code.ERROR, e.getMessage());
+        }
+      }
+
+      if (line == null || line.trim().length() == 0) {
+        return new InterpreterResult(InterpreterResult.Code.SUCCESS, "");
+      }
+
+      InterpreterResult res = livyHelper.interpret("sqlContext.sql(\"" +
+              line.replaceAll("\"", "\\\\\"")
+                  .replaceAll("\\n", " ")
+              + "\").show(" +
+              property.get("livy.spark.maxResult") + ")",
```

Comment:
is `livy.spark.maxResult` sql specific?
also generally we prefix `zeppelin.` to properties used by Zeppelin, so perhaps we could it something like `zeppelin.livy.sql.maxResult`?